x/finality/keeper/gov.go

```diff
@@ -41,28 +39,56 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 			return fmt.Errorf("the finality provider %s has voted for height %d", fpPkHex, haltingHeight)
 		}
 
+		fpBtcPk := fpPk.MustMarshal()
+		fp, err := k.BTCStakingKeeper.GetFinalityProvider(ctx, fpBtcPk)
+		if err != nil {
+			return fmt.Errorf("failed to find the finality provider %s in btcstaking: %w", fpPkHex, err)
+		}
+
+		k.Logger(ctx).Debug(
+			"fp running proposal resume finality",
+			"jailed", fp.IsJailed(),
+			"slashed", fp.IsSlashed(),
+			"height", haltingHeight,
+			"public_key", fpPkHex,
+		)
+
+		// if the FP is already jailed or slashed, no need to try to set to jail
+		// or to update the signing info
+		if fp.IsSlashed() || fp.IsJailed() {
+			continue
+		}
+
+		k.Logger(ctx).Debug(
+			"fp will be jailed",
+			"height", haltingHeight,
+			"public_key", fpPkHex,
+		)
+
 		err = k.jailSluggishFinalityProvider(ctx, fpPk)
```

Comment:
If my previous comment is true, then we should tolerate `ErrFpAlreadyJailed` and `ErrFpAlreadySlashed`

Reply:
We don't need to verify the errors anymore as the check of slashed or jailed exist
```go
		if fp.IsSlashed() || fp.IsJailed() {
			continue
		}
```